edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/metrics/AvaliabilityMetrics.cs

```diff
@@ -7,22 +7,28 @@ namespace Microsoft.Azure.Devices.Edge.Agent.Core.Metrics
     using System.IO;
     using System.Linq;
     using System.Runtime.CompilerServices;
+    using System.Threading.Tasks;
     using Microsoft.Azure.Devices.Edge.Util;
     using Microsoft.Azure.Devices.Edge.Util.Metrics;
     using Microsoft.Extensions.Logging;
 
     public interface IAvailabilityMetric
     {
         void ComputeAvailability(ModuleSet desired, ModuleSet current);
+        void OnCleanShutdown();
```

Comment:
Normally "On" prefix is used for event delegate/handler; and I think it is good enough to name as "CleanUp"